File: EasyPlan.Web/Components/Mapper/CriterionMapper.cs
﻿using EasyPlan.DomainModel.Entities;

namespace EasyPlan.Web.Components.Mapper
{
    public static class CriterionMapper
    {
        public static object Map(Criterion entity)
        {
            return new
            {
                id = entity.Id,
                title = entity.Title,
                weight = entity.Weight,
                isBenefit = entity.IsBenefit
            };
        }
    }
}

File: EasyPlan.Web/App/mappers/criterionMapper.js
define(['models/criterion', 'mappers/markMapper'], function (Criterion, markMapper) {
    return {
        map: map,
    }

    function map(src) {
        return new Criterion({
            id: src.id,
            title: src.title,
            isBenefit: src.isBenefit,
            weight: src.weight
        })
    }
});